test/stablex/stablecoin_converter.js

```diff
@@ -467,6 +466,62 @@ contract("StablecoinConverter", async (accounts) => {
       assert.equal((await stablecoinConverter.getBalance.call(user_2, erc20_2)).toString(), basicTrade.deposits[1].amount.sub(getExecutedSellAmount(fullBuyVolumes[1], prices[0], prices[1])).toString(), "Sold tokens were not adjusted correctly")
       assert.equal((await stablecoinConverter.getBalance.call(user_2, feeToken)), fullBuyVolumes[1].toString(), "Bought tokens were not adjusted correctly")
     })
+    it("Ensure fee token is burnt as expected", async () => {
+      // Fee token can't be a Mock here.
+      const TokenOWLProxy = artifacts.require("../node_modules/@gnosis.pm/owl-token/build/contracts/TokenOWLProxy")
+      const owlToken = await TokenOWL.new()
+      const owlProxyContract = await TokenOWLProxy.new(owlToken.address)
+      const owlProxy = await TokenOWL.at(owlProxyContract.address)
+      await owlProxy.setMinter(user_1)
+      const sufficientAmount = toETH(100)
+      const owlAmount = sufficientAmount.mul(new BN(accounts.length))
+
+      await owlProxy.mintOWL(user_1, owlAmount)
+
+      const stablecoinConverter = await StablecoinConverter.new(2, feeDenominator, owlProxy.address)
+      const token = await MockContract.new()
+      await owlProxy.approve(stablecoinConverter.address, owlAmount)
+      await stablecoinConverter.addToken(token.address, { from: user_1 })
+
+      // Ensure all user have sufficient feeToken
+      for (const account of accounts) {
+        await owlProxy.transfer(account, sufficientAmount, { from: user_1 })
+        await owlProxy.approve(stablecoinConverter.address, sufficientAmount, { from: account })
+      }
+
+      // First Auction
+      const tradeExample = basicTrade
+      await makeDeposits(stablecoinConverter, accounts, tradeExample.deposits)
+      const batchIndex = (await stablecoinConverter.getCurrentBatchId.call()).toNumber()
+      const orderIds = await placeOrders(stablecoinConverter, accounts, basicTrade.orders, batchIndex + 1)
+      await closeAuction(stablecoinConverter)
+
+      const solution = solutionSubmissionParams(basicTrade.solutions[0], accounts, orderIds)
+      await stablecoinConverter.submitSolution(batchIndex, solution.objectiveValue, solution.owners, solution.touchedOrderIds, solution.volumes, solution.prices, solution.tokenIdsForPrice, { from: solver })
+      await closeAuction(stablecoinConverter)
+
+      // Second Auction
```

Comment:
Should we maybe also check that applying a better solution doesn't burn the fees again?

Reply:
Yes, perhaps. Will do.